framework/components/ADatePicker/ADatePicker.js

```diff
@@ -150,15 +175,36 @@ const ADatePicker = forwardRef(
   }
 );
 
+const isValidDateTuple = (range) => {
+  const isArray = Array.isArray(range);
+  const isValidTypes = isArray && range.every(value => value instanceof Date || value === null);
+  const isValidLength = isValidTypes && range.length <= 2;
+  return isArray && isValidDateTuple && isValidLength;
+};
+
+function rangeTupleValidator(propValue, key, componentName) {
+  if (!isValidDateTuple(propValue)) {
+    throw new Error(
+      "Invalid prop 'value' supplied to '" + componentName + "'. "
+      + "When using a range, pass a tuple indicting a start and end "
+      + "date, or 'null' if empty."   
+    )
+  }
+  return null;
+}
+
 ADatePicker.propTypes = {
   /**
    * Handles the `change` event for when a date is selected.
    */
   onChange: PropTypes.func,
   /**
-   * Sets the date selected.
+   * Sets the date(s) selected.
    */
-  value: PropTypes.instanceOf(Date)
+  value: PropTypes.oneOfType([
+    PropTypes.instanceOf(Date),
+    PropTypes.arrayOf(rangeTupleValidator),
```

Comment:
```PropTypes.instanceOf(rangeTupleValidator),```

Reply:
I think this should actually stay as `oneOfType` to differentiate the single date selection (`instanceOf(Date)`) and the range selection (`arrayOf(rangeTupleValidator)`). When selecting a single date on the picker, only the `Date` object needs to be passed. When selecting a range, it should contain an array that contains either `null` or `Date` objects. That is a bit too granular for the default PropTypes validators, so I created the custom `rangeTupleValidator` that PropTypes specifies you can pass to the `arrayOf` function. Here is a good example from the [PropTypes docs](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes):

```js
// You can also supply a custom validator to `arrayOf` and `objectOf`.
// It should return an Error object if the validation fails. The validator
// will be called for each key in the array or object. The first two
// arguments of the validator are the array or object itself, and the
// current item's key.
customArrayProp: PropTypes.arrayOf(function(propValue, key, componentName, location, propFullName) {
if (!/matchme/.test(propValue[key])) {
    return new Error(
    'Invalid prop `' + propFullName + '` supplied to' +
    ' `' + componentName + '`. Validation failed.'
    );
}
})
```